modules/mobile/spec/request/appointments_vaos_v2_list_request_spec.rb

```diff
@@ -210,5 +214,74 @@
         expect(uniq_statuses).to eq(%w[CANCELLED BOOKED SUBMITTED])
       end
     end
+
+    describe 'provider data' do
+      before do
+        mock_facility
+        mock_clinic
+      end
+
+      let(:start_date) { Time.zone.parse('1991-01-01T00:00:00Z').iso8601 }
+      let(:end_date) { Time.zone.parse('2023-01-01T00:00:00Z').iso8601 }
+
+      def fetch_appointments
+        VCR.use_cassette('appointments/VAOS_v2/get_appointments_with_mixed_provider_types',
+                         match_requests_on: %i[method uri]) do
+          VCR.use_cassette('providers/get_provider_200', match_requests_on: %i[method uri], tag: :force_utf8) do
+            get '/mobile/v0/appointments', headers: iam_headers, params: params
+          end
+        end
+      end
+
+      context 'when upstream appointments index returns provider names' do
+        it 'adds names to healthcareProvider field' do
+          fetch_appointments
+          # it would be better to find by id, but there's currently a bug that's causing these ids to be nil
```

Comment:
When andrew recorded the "all of ruben's appointments" vcr, many of the appointments didn't have ids. It seems that they do now. After this ticket, I will check staging and update the VCRs if they're substantially different, which will hopefully reveal any issues in the proxy as a result of those data changes.